staging_service/import_specifications/file_parser.py

```diff
@@ -166,14 +166,58 @@ def __post_init__(self):
 
 def parse_import_specifications(
     paths: tuple[Path],
-    file_type_resolver: Callable[[Path], FileTypeResolution]
+    file_type_resolver: Callable[[Path], FileTypeResolution],
+    log_error: Callable[[Exception], None]
 ) -> ParseResults:
     """
     Parse a set of import specification files and return the results.
 
     paths - the file paths to open.
     file_type_resolver - a callable that when given a file path, returns the type of the file or
         a parser for the file.
+    log_error - callable for logging an exception.
```

Comment:
You could use a named tuples here instead of regular tuples and it may end up more readable.

Reply:
I'm not sure which tuples you're talking about - there's only the `paths` tuple which only contains an indeterminate number of Path objects so a named tuple doesn't seem to make sense

Reply:
This does make me realize I'm not annotating tuples correctly. I'll make a new PR and we can continue the discussion there if necessary